components/script/dom/location.rs

```diff
@@ -57,7 +60,10 @@ impl LocationMethods for Location {
 
     // https://html.spec.whatwg.org/multipage/#dom-location-reload
     fn Reload(&self) {
-        self.window.load_url(self.get_url(), true, None);
+        self.window.main_thread_script_chan().send(
+            MainThreadScriptMsg::Navigate(self.window.pipeline_id(),
+                                          LoadData::new(self.get_url(), None, Some(self.get_url())),
+                                          true)).unwrap();
```

Comment:
I would prefer to add an argument to `load_url` that indicates that we should ignore the fragment check.